.github/workflows/release.yml

```diff
@@ -1,10 +1,8 @@
-name: release workflow
+name: deploy to environments
 
 on:
-  push:
-    branches: [main]
-    tags: [v**]
-
+  workflow_dispatch:
+    
 env:
   BRANCH_NAME: ${{ github.event.ref.BRANCH_NAME }}
```

Comment:
This makes me doubt my comment above, since I assume this workflow works as intended. how come we use this?